### DISCUSSION TITLE
facebook login in localhost

### DISCUSSION THREAD

**Discussion by Marius555** [pocketbase/pocketbase#3469] [Q&A]
im trying to integrate facebook login in oAuth2, but facebook keep asking me for https to work....how to solve this issue? i just started my app and i already need to have my own domain and SSL certificate? its cant be this hard only for facebook login....

**Comment by ganigeorgiev** [comment]
This is a Facebook (and some other OAuth2 providers) requirement, I don't think there is anything we can do on the PocketBase side.

You can use [ngrok](https://ngrok.com/download) or other similar tool to create a public address for your localhost:

```sh
./pocketbase serve

# in another terminal run
ngrok http 8090

# use the generated url...
```

**Reply by Marius555** [reply]
ya, so i tried, but here is the problem....you can use "ngrok http 8090" the problem is that facebook still doesnt allow this, because request comes from localhost:3000 and you cant only use one ngrok session per free plan

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean.

If you use the all-in-one OAuth2 flow, for the redirect url you need to specify only the PocketBase address (`https://your-ngrok-url.com/api/oauth2-redirect`).

If you use the Manual code exchange flow, eg. the redirect handler is in your node app, then you need to start ngrok with the port for your node application and use your node endpoint address as redirect url during the OAuth2 app creation.

**Reply by Marius555** [reply]
im really confused... i set redirect url in faceebok as ngrok https link which redirects to "localhost:8090" in pocketbase flow its says i have to redirect to "http://127.0.0.1:8090/api/oauth2-redirect", is it enough to redirect only to "localhost:8090"? them pop up window from facebook comes and say "Insecure Login Blocked You can't get an access token or login to this app from an insecure page. Try re-loading the page as https://"

**Reply by ganigeorgiev** [reply]
If you use ngrok, you have to replace the `http://127.0.0.1:8090` part with the generated url.

**Reply by Marius555** [reply]
yes i did it, my redirect url in facebook is set to "https://4cc8-86-100-153-65.ngrok-free.app"

**Comment by lucas-0liveira** [comment]
Integrating Facebook login with OAuth2 requires a secure connection, which means using HTTPS. This is a security measure to protect user data during the authentication process.

To address this issue, you will indeed need to have your own domain and an SSL certificate. Here are the steps you can take:

Obtain an SSL Certificate:

You can obtain an SSL certificate through a Certificate Authority (CA) or consider using a service like Let's Encrypt, which offers free SSL certificates.
Configure HTTPS on Your Server:

Install the SSL certificate on your server. This process might vary depending on your hosting provider or server setup. Many hosting platforms have guides or support resources for this.
Update Your App Settings on Facebook Developer:

Go to the Facebook Developer Dashboard and navigate to your app's settings.
In the "Valid OAuth Redirect URIs" section, make sure you have added the HTTPS version of your callback URL.
Update Your Code:

Ensure that your application's code is configured to use the HTTPS version of your domain for the OAuth2 flow.
Test the Integration:

Test the Facebook login functionality in your application to ensure it's working correctly with HTTPS.
While it may seem complex at first, ensuring a secure connection is crucial for protecting user data. Additionally, having HTTPS in place is a standard practice for web applications.

If you're just starting out and don't have your own domain yet, you might consider using a testing domain with HTTPS support for development purposes until you're ready to deploy in a production environment.

Remember, this requirement is not specific to Facebook but is a general best practice for secure web development.

**Reply by ganigeorgiev** [reply]
> To address this issue, you will indeed need to have your own domain and an SSL certificate.

_That's not exactly true. For test purposes ngrok or other tunneling service should suffice. The above also seems like generated from ChatGPT, if that's the case please avoid doing it._

**Comment by gaetan1903** [comment]
in Facebook login parameters, you can deactivate "Only HTTPS"

![image](https://github.com/pocketbase/pocketbase/assets/43904633/d278f225-98d1-49c1-a4e5-a3c3abf78918)

**Reply by JorSoi** [reply]
not anymore apparently

**Reply by RashnerdUX** [reply]
Just came across this problem. Makes no sense. How do they expect one to test it
